examples/vision/cutmix.py

```diff
@@ -0,0 +1,367 @@
+# -*- coding: utf-8 -*-
+"""
+Title: CutMix data augmentation for image classification
+Author: [Sayan Nath](https://github.com/sayannath)
+Date created: 2021/06/08
+Last modified: 2021/06/08
+Description: Data augmentation with CutMix for image classification on CIFAR-10.
+"""
+
+"""
+## Introduction
+"""
+
+"""
+_CutMix_ is a data augmentation technique that addresses the issue of information loss and inefficiency present in regional dropout strategies. Instead of removing pixels and filling them with black or grey pixels or Gaussian noise, you replace the removed regions with a patch from another image, while the ground truth labels are mixed proportionally to the number of pixels of combined images. CutMix was proposed in [CutMix: Regularization Strategy to Train Strong Classifiers with Localizable Features](https://arxiv.org/pdf/1905.04899.pdf) (Yun et al., 2019)
+
+It's implemented with the following formulas:
+![](https://i.imgur.com/cGvd13V.png)
+where M is the binary mask which indicates the cutout and the fill-in regions from the two randomly drawn images and λ is drawn from [Beta(α,α) distribution](https://en.wikipedia.org/wiki/Beta_distribution) and `λ ∈ [0, 1]`
+
+The coordinates of bounding boxes are ![](https://i.imgur.com/eNisep4.png) which indicates the cutout and fill-in regions in case of the images.
+The bounding box sampling is represented by:
+
+![](https://i.imgur.com/Snph9aj.png)
+where `rx,ry` are randomly drawn from a uniform distribution with upper bound
+
+This example requires TensorFlow 2.4 or higher.
+"""
+
+"""
+## Import the necessary packages
+"""
+
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+import tensorflow as tf
+from tensorflow import keras
+
+np.random.seed(42)
+tf.random.set_seed(42)
+
+"""
+## Load the CIFAR-10 dataset
+
+In this example, we will use the
+[CIFAR- 10 dataset](https://www.cs.toronto.edu/~kriz/cifar.html).
+"""
+
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()
+
+
+y_train = tf.keras.utils.to_categorical(y_train, num_classes=10)
+y_test = tf.keras.utils.to_categorical(y_test, num_classes=10)
+
+print(x_train.shape)
+print(y_train.shape)
+
+print(x_test.shape)
+print(y_test.shape)
+
+class_names = [
+    "Airplane",
+    "Automobile",
+    "Bird",
+    "Cat",
+    "Deer",
+    "Dog",
+    "Frog",
+    "Horse",
+    "Ship",
+    "Truck",
+]
+
+"""
+## Define hyperparameters
+"""
+
+AUTO = tf.data.experimental.AUTOTUNE
+BATCH_SIZE = 32
+IMG_SHAPE = 32
```

Comment:
Call this IMG_SIZE. IMG_SHAPE would be like (32, 32).

Reply:
Ya Sure! Will do that